Add support for `quote_field_suffix` to `simple_query_string`.

jpountz opened the pull request:
Closes #18641

nik9000 commented on docs/reference/how-to/recipes.asciidoc:
```diff
@@ -0,0 +1,225 @@
+stemming into account?
+
+Fortunately, the `query_string` query has a feature that allows to solve exactly
+this problem: `quote_field_suffix`. It allows to tell Elasticsearch that words
```
I'm uncomfortable suggesting that anyone expose `query_string` to end users. Its parsing is too brittle for this to be a good idea. I know, I've done it and worked myself into a local minima by adding dozen regexes around user's regexes to prevent it from crashing.

jpountz replied:
Would you like it better if we added support for `quote_field_suffix` to `simple_query_string` first (it looks easy to do) and then used `simple_query_string` in those examples?

nik9000 replied:
That'd be awesome! I'm fine with documenting the parameter for `query_string` but I think we should avoid telling a story about using it.

jpountz commented:
@jimferenczi @nik9000 I added support for `quote_field_suffix` to `simple_query_string` and fixed the docs to use `simple_query_string` rather than `query_string` as an example. Would you ming taking another look?
